frontends/web/src/contexts/WCWeb3WalletProvider.tsx

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright 2023 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ReactNode, useEffect, useState } from 'react';
+import WCWeb3WalletContext from './WCWeb3WalletContext';
+import { IWeb3Wallet, Web3Wallet } from '@walletconnect/web3wallet';
+import { Core } from '@walletconnect/core';
+
+type TProps = {
+    children: ReactNode;
+  }
+
+export const WCWeb3WalletProvider = ({ children }: TProps) => {
+  const [web3wallet, setWeb3wallet] = useState<IWeb3Wallet>({} as IWeb3Wallet);
```

Comment:
initially it is undefined, it would be saver to remove `{} as IWeb3Wallet` and check if `web3wallet` is not undefined .